shell/common/shell_test_external_view_embedder.cc

```diff
@@ -64,7 +64,9 @@ SkCanvas* ShellTestExternalViewEmbedder::CompositeEmbeddedView(int view_id) {
 void ShellTestExternalViewEmbedder::SubmitFrame(
     GrDirectContext* context,
     std::unique_ptr<SurfaceFrame> frame) {
-  frame->Submit();
+  if (frame) {
```

Comment:
If the frame pointer is null (checked later), there must be an early return so the submitted frame count in not bumped up.

So something like.

```
if (!frame) {
  return;
}
```

followed by the removal of `frame &&` later on like 70 since that check will now be redundant.